source/common/ratelimit/ratelimit_impl.cc

```diff
@@ -86,7 +92,16 @@ GrpcFactoryImpl::GrpcFactoryImpl(const envoy::config::ratelimit::v2::RateLimitSe
 }
 
 ClientPtr GrpcFactoryImpl::create(const absl::optional<std::chrono::milliseconds>& timeout) {
-  return std::make_unique<GrpcClientImpl>(async_client_factory_->create(), timeout);
+  // TODO(junr03): legacy rate limit is deprecated. Remove support for the lyft proto after 1.7.0.
+  if (use_data_plane_proto_) {
+    return std::make_unique<GrpcClientImpl>(
+        async_client_factory_->create(), timeout,
+        "envoy.service.ratelimit.v2.RateLimitService.ShouldRateLimit");
+  }
+  ENVOY_LOG_MISC(warn, "legacy rate limit client is deprecated, update your service to support "
```

Comment:
When does this run? Only on startup? Or on every request?

Reply:
good point. I moved to the GrpcFactoryImpl constructor which is called only once in the server code https://github.com/envoyproxy/envoy/blob/08c139994d97d0513d6f305d17825cd4e74daece/source/server/configuration_impl.cc#L83